corehq/apps/case_search/filter_dsl.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+from django.utils.translation import ugettext as _
+from eulxml.xpath.ast import Step, serialize
+from six import integer_types, string_types
+
+from corehq.apps.es import filters
+from corehq.apps.es.case_search import (
+    CaseSearchES,
+    case_property_range_query,
+    exact_case_property_text_query,
+    reverse_index_case_query,
+)
+
+
+class CaseFilterError(Exception):
+
+    def __init__(self, message, filter_part):
+        self.filter_part = filter_part
```

Comment:
planning on using this later when showing errors to the user. It will be something like "the part of your query that failed was "{filter_part}""